docker_builder/main.tf

```diff
@@ -0,0 +1,49 @@
+terraform {
+  required_providers {
+    null = "~> 2.1"
+    aws  = ">= 2.62"
+  }
+}
+
+module "git_branch" {
+  source  = "matti/resource/shell"
+  command = "git rev-parse --abbrev-ref HEAD"
+}
+
+module "git_sha" {
+  source  = "matti/resource/shell"
+  command = "git rev-parse HEAD"
+}
+
+locals {
+  registry   = "${var.repository}/${var.service}"
+  args       = join(" ", var.args)
+  git_branch = lower(replace(module.git_branch.stdout, "/", "-"))
+  git_sha    = module.git_sha.stdout
+}
+
+resource "null_resource" "build_image" {
+  triggers = {
+    image_tag = module.git_sha.stdout
+  }
+
+  provisioner "local-exec" {
+    command = <<EOT
+      aws ecr get-login-password \
+        --region ${var.region} \
```

Comment:
will need to check is we need to ask for `region` input or it will default to what present in the AWS configuration

Reply:
I would assume anyone building terraform is going to supply it like so:

```
provider "aws" {
  version = "~> 2.62"
  region  = var.region
}
```

But I'm not sure this will set region for `aws` command line code too, or just for the terraform module.

Reply:
Nope 😞 
will need to add it to `variables.tf`